### PR TITLE
[IMP] account: exchange rate entry prepare vals

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
Just a prepare method in order to be able to include more fields in the exchange rate journal items.
This is needed to add information in the journal items such as stock move,  purchase order, sale order. This is useful in systems that do automatic reconciliation based on purchase or sales. The exchange journal items should be taken into account in this case.

cc @ForgeFlow

### CODE REVIEW

**Comment by aliyatastemirova** [comment]
Hi, thank you :slightly_smiling_face: 
It seems that the email you used in the commit doesn't match the one in the cla. The email used here is `aaron.henriquez@forgeflow.com` and in the cla it's `ahenriquez@forgeflow.com`.
Also, do we need this in 15.0? Could you please give a bit more information about why? Thanks!

File: addons/account/models/account_move.py
Comment (MiquelRForgeFlow) [inline addons/account/models/account_move.py:5252]:
```suggestion
                exchange_vals_0 = self._prepare_exchange_vals_0(line, sequence, exchange_line_account)
```
Although not used, I think you could pass the exchange, just in case some customization needs it.

**Comment by AaronHForgeFlow** [comment]
@aliyatastemirova , I just completed a bit the commit message. Is that enough? Also updated my the email in the CLA. 

Basically I want to add the purchase order and sale order information to the exchange rate journal entries. The process is the following:
* Purchase/Sale orders that are received and invoiced along a long period and at the end the product interim accounts are unbalanced
* I want to reconcile the interim account at order level, so I do a write-off by order line
* The reconciliation could trigger a currency exchange rate entry and I want to have it linked to the PO to ensure the order is correctly reconciled and also for reporting purposes.

The design is quite different in v16 and master branches, a different prepare method would be needed in those versions. I need this in v15 but if you think this is useful I can propose the equivalent code in master.

**Comment by aliyatastemirova** [comment]
@AaronHForgeFlow thank you. To me, this is fine, but we need to wait for a review from @smetl :slightly_smiling_face:

File: addons/account/models/account_move.py
Comment (smetl) [inline addons/account/models/account_move.py:5519]: You probably want to have a single method for both lines. In case you need to search for others business models, you only want to do it once

File: addons/account/models/account_move.py
Comment (smetl) [inline addons/account/models/account_move.py:5516]:
```suggestion
        }
```

File: addons/account/models/account_move.py
Comment (smetl) [inline addons/account/models/account_move.py:5528]:
```suggestion
        }
```

**Review by smetl — CHANGES_REQUESTED** [review]
In your commit message, you are talking about link with PO/SO/stock moves but in all cases, the exchange difference should not be part of it I think. It's a pure accounting mechanism. Can you give me a concrete example of what you are trying to achieve here plz?

**Comment by AaronHForgeFlow** [comment]
@smetl @aliyatastemirova thank you for considering this PR. Actually, I am having currency exchange rate entries when reconciling the stock interim accounts. This is because I am reconciling them by purchase order. This happens for POs in a  different currency than the company currency, When doing that sometimes I have to create a write-off so it is balanced. The write-off is in the same currency than the PO currency, but for some reason it triggers the currency exchange rate journal entry creation. Actually, what I want is to prevent this journal entry to be created in this case, as long as I am reconciling journals in the same currency. I will update the PR according to achieve that.

**Comment by AaronHForgeFlow** [comment]
It seems the issue commented is not happening anymore in v15, so I am closing the PR and I will reopen in case new findings are  detected.

**Pull request closed without merging**
